docker/docker-compose.yml

```
@@ -0,0 +1,88 @@
+name: shutter-explorer
+services:
+  observer:
+    build:
+      context: ../observer/
+      dockerfile: ../observer/Dockerfile
+    restart: unless-stopped
+    depends_on:
+      - db
+    env_file: .env
+    environment:
+        - DB_HOST=db
+        - DB_PORT=5432
+        - DB_USER=postgres
+        - DB_PASSWORD=${OBSERVER_DB_PASSWORD}
+        - DB_NAME=postgres
+        - DB_SSL_MODE=disable
+        - P2P_BOOTSTRAP_ADDRESSES=${P2P_BOOTSTRAP_ADDRESSES}
+        - P2P_ENVIRONMENT=${P2P_ENVIRONMENT}
+        - P2P_DISCOVERY_NAMESPACE=${P2P_DISCOVERY_NAMESPACE}
+    command: ["start", "--rpc-url", "${RPC_URL}", "--beacon-api-url", "${BEACON_API_URL}", "--sequencer-contract-address", "${SEQUENCER_CONTRACT_ADDRESS}", "--validator-registry-contract-address", "${VALIDATOR_REGISTRY_CONTRACT_ADDRESS}", "--p2pkey", "${P2P_KEY}"]
+
+  db:
+    image: postgres:14.12
+    restart: unless-stopped
+    env_file: .env
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=${OBSERVER_DB_PASSWORD}
+      - POSTGRES_DB=postgres
+    volumes:
+      - ./pgdata:/var/lib/postgresql/data
+    ports: 
+      - "5432:5432"
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      start_interval: "2s"
+      start_period: "30s"
+      interval: "30s"
+      timeout: "5s"
+      retries: 3
+
+  backend:
+    build: 
+      context: ../backend/
+      dockerfile: ../backend/Dockerfile
+    restart: unless-stopped
+    depends_on: 
+      - observer
+    env_file: .env
+    environment:
+      - SERVER_PORT=5000
+      - OBSERVER_DB_HOST=db
+      - OBSERVER_DB_PORT=5432
+      - OBSERVER_DB_USER=postgres
+      - OBSERVER_DB_PASSWORD=${OBSERVER_DB_PASSWORD}
+      - OBSERVER_DB_NAME=postgres
+      - OBSERVER_DB_SSL_MODE=disable
+      - ERPC_DB_HOST=${ERPC_DB_HOST}
+      - ERPC_DB_PORT=${ERPC_DB_PORT}
+      - ERPC_DB_USER=${ERPC_DB_USER}
+      - ERPC_DB_PASSWORD=${ERPC_DB_PASSWORD}
+      - ERPC_DB_NAME=${ERPC_DB_NAME}
+      - ERPC_DB_SSL_MODE=${ERPC_DB_SSL_MODE}
+      - GENESIS_TIMESTAMP=${GENESIS_TIMESTAMP}
+      - SLOT_DURATION=${SLOT_DURATION}
+      - SLOTS_PER_EXPOCH=${SLOTS_PER_EXPOCH}
+    command: ["./backend"]
+    labels:
+      caddy: "${SERVICE_DOMAIN_NAME}"
+      caddy.@browser: "method GET"
+      caddy.redir: "@browser https://blog.shutter.network/shutterized-gnosis-chain-is-now-live/"
+      caddy.reverse_proxy: "{{ upstreams 5000 }}"
+      caddy.header.Access-Control-Allow-Origin: "*"
+      caddy.header.Access-Control-Allow-Methods: "GET, POST, OPTIONS"
+      caddy.header.Access-Control-Allow-Headers: "Content-Type, Authorization"
+
+  caddy:
+    image: lucaslorentz/caddy-docker-proxy:latest
+    restart: always
+    volumes:
+      - ${DATA_DIR:-./data}/caddy:/data
+      - /var/run/docker.sock:/var/run/docker.sock
+    ports:
+      - 80:80
+      - 443:443
```

I only recently learned myself, that it makes sense to also include 
```suggestion
      - "443:443/udp"
```
Since HTTP 2, 3 and QUICK all like to use udp.